diff --git zookld.c zookld.c
index 0d60857..55b9543 100644
--- zookld.c
+++ zookld.c
@@ -144,30 +144,38 @@ pid_t launch_svc(CONF *conf, const char *name)
                     break;
     }
 
+    /*
+     * Below, we extract the parameters that govern the privileges with
+     * which the service will run: userid, groupid, group list, the
+     * chroot directory. Your job is to actually set those privileges.
+     * What constraints are there on the order in which those privileges
+     * should be set?
+     */
+
+    /* you will change the real, effective, and saved userid to 'uid'. */
     if (NCONF_get_number_e(conf, name, "uid", &uid))
     {
-        /* change real, effective, and saved uid to uid */
         warnx("setuid %ld", uid);
     }
 
+    /* you will change the real, effective, and saved groupid to 'gid'. */
     if (NCONF_get_number_e(conf, name, "gid", &gid))
     {
-        /* change real, effective, and saved gid to gid */
         warnx("setgid %ld", gid);
     }
 
+    /* you will set the grouplist, based on gids. */
     if ((groups = NCONF_get_string(conf, name, "extra_gids")))
     {
         ngids = 0;
         CONF_parse_list(groups, ',', 1, &group_parse_cb, NULL);
-        /* set the grouplist to gids */
         for (i = 0; i < ngids; i++)
             warnx("extra gid %d", gids[i]);
     }
 
+    /* you will chroot to dir. */
     if ((dir = NCONF_get_string(conf, name, "dir")))
     {
-        /* chroot into dir */
     }
 
     signal(SIGCHLD, SIG_DFL);
